### PR TITLE
New array Extension plus tests

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# XRTK - Mixed Reality Toolkit Change Request

## Overview

* Added new "InitialiseArray" array extension to set all values of an array to a new value, including Tests
* Added null check on Array AddItem
* Fixed license in Core test 2

### CODE REVIEW

File: XRTK-Core/Assets/XRTK.Tests/Core/TestFixture_03_ExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using NUnit.Framework;
+using UnityEngine;
+using XRTK.Extensions;
+
+namespace XRTK.Tests.Core
+{
+    public class TestFixture_03_ExtensionsTests
+    {
+        [Test]
+        public void Test_01_Array_AddItem_Extension()
```
Comment (StephenHodgson) [inline XRTK-Core/Assets/XRTK.Tests/Core/TestFixture_03_ExtensionsTests.cs]: Can we also test to add the item at an random index and ensure that it's inserted in the correct postion?

File: XRTK-Core/Packages/com.xrtk.core/Extensions/ArrayExtensions.cs [hunk 2]
```diff
@@ -37,12 +37,43 @@ public static bool IsValidArray(this Array array)
         /// <param name="array">The array to extend</param>
         /// <param name="newItem">The item to add to the array</param>
         /// <returns></returns>
-        public static T[] AddItem<T>(this T[] array, T newItem, int insertAtIndex = 0)
+        public static T[] AddItem<T>(this T[] array, T newItem)
         {
+            // Initialise the array if it is null
+            if (array == null)
+            {
+                array = new T[0];
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Extensions/ArrayExtensions.cs]: Take a look at my changed in [455](https://github.com/XRTK/XRTK-Core/pull/455/files?file-filters%5B%5D=.cs&hide-deleted-files=true#diff-d33dfd0c48086e9a05bb9888d60a437bR45)
```suggestion
                return new[] { newItem };
```

File: XRTK-Core/Packages/com.xrtk.core/Extensions/ArrayExtensions.cs [hunk 3]
```diff
@@ -37,12 +37,43 @@ public static bool IsValidArray(this Array array)
         /// <param name="array">The array to extend</param>
         /// <param name="newItem">The item to add to the array</param>
         /// <returns></returns>
-        public static T[] AddItem<T>(this T[] array, T newItem, int insertAtIndex = 0)
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Extensions/ArrayExtensions.cs]: I don't mind keeping this if we're actually inserting it correctly. ATM it doesn't actually insert it at the right index, which is a bug, and why I modified it in the other PR I was working on.

File: XRTK-Core/Assets/XRTK.Tests/Core/TestFixture_03_ExtensionsTests.cs (hunk 1 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Assets/XRTK.Tests/Core/TestFixture_03_ExtensionsTests.cs]: This is not an option it currently supports. Might add that in the future

File: XRTK-Core/Packages/com.xrtk.core/Extensions/ArrayExtensions.cs (hunk 3 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Extensions/ArrayExtensions.cs]: We removed the insert at index option for now

**Comment by SimonDarksideJ** [comment]
Applied suggested fix to both extensions, if the array is null, return a single item array with the item provided

**Pull request merged**